beacon_node/network/src/sync/range_sync/chain.rs

```diff
@@ -396,6 +396,9 @@ impl<T: BeaconChainTypes> SyncingChain<T> {
                     self.request_batches(network)?;
                 }
             }
+        } else if !self.good_peers_on_custody_subnets(self.processing_target, network) {
+            // If there's no good custody peers for this epoch, batch won't be created
+            return Ok(KeepChain);
```

Comment:
Why is this condition necessary?

Reply:
This is the key change of this PR.

The current behaviour on startup:
- `send_batch` sends both range requests
- Column range request may fail immediately with `NoCustodyPeers`
- It will then immediately retry with `send_batch`, spamming `BlocksByRange`requests to peers
- At some point it probably either reached its attempt limit or get rate limited by peer.

This change prevent sending out range requests until we have at least one peer across all custody subnets.

This is now working on a local devnet and we're able to range sync from a Lighthouse supernode 🎉

Reply:
There are a few things to fix though:
- We're sampling too fast during sync - we _could_ bump up the limit for now, but we have no throttling at all, and I just see logs scrolling through so we may be DOSing the peer. I'll have to check the self rate limiter config to fix this.
- `good_peers_on_custody_subnets` implementation can be improved to include fallbcaks and using MetaData, but I think the current version would be fine to roll out.

I think once we fix the sampling issue, it would be great to deploy to devnet, and we can continue to iterate from there.

Reply:
Oh sorry I thought this was the other KeepChain!

This is to handle the case where no batches was sent for the current processing target, which ends up in the error below.

This is now a valid state when we have no custody peers, but we still want to keep the existing condition to return an error if no batch was sent for some other reasons.

Reply:
Updated code comment in ede964ee